contractcourt/channel_arbitrator.go

```diff
@@ -3117,6 +3181,9 @@ func (c *ChannelArbitrator) handleBlockbeat(beat chainio.Blockbeat) error {
 		}
 	}
 
+	// Launch all active resolvers when a new blockbeat is received.
```

Comment:
I think we should not do this, it feels hacky.

Reply:
how is this hacky? Or what is the alternative?

Reply:
yes I agree that we should always launch the 2 contest resolver, for the other resolvers it does imo not really make sense because they will already be marked as resolved.

Reply:
> So maybe we can funnel in the blockbeat notification to the resolver and then can trigger the correct timeout/success Resolver in that way ?

Not sure if this is made before or after our sync call, but just to clarify it again here - the current design is a tradeoff of the scope of change and functionality in place. We cannot easily make the resolvers as blockbeat consumers, nor we can make them oblivious of block notifications, as they require more refactoring.